laundry/views.py

```diff
@@ -0,0 +1,157 @@
+import calendar
+import datetime
+
+from django.utils import timezone
+from django.utils.timezone import make_aware
+from requests.exceptions import HTTPError
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from laundry.api_wrapper import all_status, hall_status
+from laundry.models import LaundryRoom, LaundrySnapshot
+from laundry.serializers import LaundrySnapshotSerializer
+
+
+class Halls(APIView):
+    def get(self, request):
+        try:
+            return Response(all_status())
+        except HTTPError:
+            return Response({"error": "The laundry api is currently unavailable."}, status=503)
+
+
+class HallInfo(APIView):
+    def get(self, request, hall_id):
+        try:
+            return Response(hall_status(int(hall_id)))
+        except ValueError:
+            return Response({"error": "Invalid hall id passed to server."}, status=404)
```

Comment:
This could be simplified into a LaundryRoom [get_or_404](https://docs.djangoproject.com/en/3.1/topics/http/shortcuts/#get-object-or-404) and then `hall_status` could just take in a LaundryRoom object. That way we move away from just passing ints around and hoping a room exists with that id